File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    remaining_percent = []
    for i in range(len(progresses)):
        remaining_percent.append(((100 - progresses[i]) // speeds[i]) + ((100 - progresses[i]) % speeds[i] != 0))
    answer = []
    while remaining_percent:
        current_progress = remaining_percent.pop(0)
        cnt = 1
        while remaining_percent and current_progress >= remaining_percent[0]:
            remaining_percent.pop(0)
            cnt += 1
        answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> remaining_days;
    
    for (int i = 0; i < progresses.size(); i++) {
        int days = ((100 - progresses[i]) / speeds[i]) + ((100 - progresses[i]) % speeds[i] != 0);
        remaining_days.push(days);
    }

    while (!remaining_days.empty()) {
        int cur = remaining_days.front(); remaining_days.pop();
        int cnt = 1;
        while (!remaining_days.empty() && cur >= remaining_days.front()) {
            remaining_days.front(); remaining_days.pop();
            cnt++;
        }
        answer.push_back(cnt);
    }
    return answer;
}